### PR TITLE
M8: parcel-ID generation + scan write engine + LABEL_GENERATED 

### PR DESCRIPTION
**Pull request by BanerjeeAgniva**
# Pull Request

## Summary

M8 **PR2 of 4** — parcel-ID generation + the append-only scan **write engine** + the M4 `LABEL_GENERATED` seam. This is the first PR that actually **writes** to the ledger PR1 built.

**Business moment:** a DA reaches a customer's door, taps "Generate Label", a sticker prints — `1DD-BOM-260711-000001` — and that barcode becomes the parcel's identity for its whole journey. On the customer's tracking page the id appears. Technically: the DA app calls M8, M8 mints the string under a per-hub/day counter lock, writes the `LABEL_GENERATED` scan, and (after commit) emits a `ScanEvent` that M4 uses to stamp `shipments.parcel_id`.

## What Changed

**`common`** — `ScanEvent` gains `parcelId` + `occurredAt` (D-005); tolerant reader intact + a 2-arg convenience ctor for PR3/PR4's non-label scans. Nothing constructed it before, so no breakage.

**`barcode`** — the write engine + label path:
- `ScanLedgerService`/`Impl` — the **write engine, built once** (reused by PR3/PR4): idempotent on `clientScanId`, insert, then raise an in-process `ScanRecorded`.
- `ParcelIdGenerator` — `1DD-{IATA}-{yyMMdd}-{seq6}` under the counter's `SELECT FOR UPDATE`.
- `LabelService`/`Impl` — **idempotent per shipment**: a retry returns the same barcode and does *not* burn a counter value.
- `ScanEventProducer` — `@TransactionalEventListener(AFTER_COMMIT)`; broadcasts only types that are a `ScanEventType` (van scans fall through, D-004).
- `BarcodeMessagingConfig` — declares the `oneday.scan.events` exchange.
- `ScanController` — `POST /api/v1/scan/label`. pom: +amqp, +validation.

**`orders`** — `ScanEventsConsumer` stamps `shipments.parcel_id` on `LABEL_GENERATED` (no state transition); closes the pre-existing TODO.

## Why This Change?

Every remaining M8 seam writes through `ScanLedgerService`, so building it once here (idempotent, event-after-commit) means PR3 (REST lifecycle scans) and PR4 (van custody) are thin doors over a proven engine.

## Testing
- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

### Test Evidence

Unit (11 new): parcel-ID format + counter bump; label idempotency (existing → no re-mint); producer broadcast-vs-skip; `ScanEvent` JSON round-trip (jsr310 + snake_case); **orders consumer stamps `parcel_id` / other scans transition without stamping**. Full `common,barcode,orders` suites green.

**Live boot-verify** against the dev DB (Flyway applied `V8_1`/`V8_2`):
```
POST /api/v1/scan/label {shipment_id, dest_city:"BOM"}  → 1DD-BOM-260711-000001
re-POST (new idem key)                                   → SAME barcode (idempotent)
scan_ledger: exactly 1 row (LABEL_GENERATED, DA), parcel_id set
parcel_id_counter: (BOM, today) next_seq = 2  (bumped once, not twice)
ScanEventProducer: "Publishing ScanEvent LABEL_GENERATED …" (AFTER_COMMIT fired)
```

> **Note on the async M4 stamp:** in the live run the `ScanEvent` reached the DLQ instead of stamping `shipments.parcel_id`. Root cause is **environmental, not this code**: a remote/deployed instance running pre-PR2 code is connected to the shared CloudAMQP and consuming `orders.scan` (confirmed — `orders.scan` keeps a consumer with no local app running), and it dead-letters the extended `ScanEvent` it can't handle. The producer side is live-verified, the payload round-trips, and the consumer stamp is unit-proven — so the hop works on an isolated broker / once PR2 is deployed everywhere.

### Areas Affected
- [x] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [x] Database
- [ ] NA

### Modules Affected
- [ ] M1 — Auth
- [ ] M2 — Pricing
- [ ] M3 — Grid
- [x] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [ ] M7 — Hub
- [x] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

## Risks / Concerns

- **`destCity` rides the request** (no cross-module shipment-lookup port) — safe because per D-002 the barcode's hub segment is human-convenience, never parsed for routing. Marked with a `ponytail:` note. Also surfaced: `shipments.dest_city` stores city **names** ("MUMBAI"), not IATA — the DA app must supply the 3-letter code (worth a follow-up to standardise).
- **Auth gating deferred to PR3** (per plan) — `actorId` rides the request for now.
- **`common` change** — one field-add to `ScanEvent`; tolerant reader means existing consumers don't break, but the deployed app should be updated (see the async-stamp note).

## Checklist
- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: common/src/main/java/com/oneday/common/kafka/events/ScanEvent.java [hunk 1]
```diff
@@ -4,17 +4,24 @@
...
  * Inbound event consumed by M4 from {@code oneday.scan.events} (produced by M8).
  *
- * <p>Minimal consumption contract — see {@link DaEvent} for the tolerant-reader rationale.
- * Note {@code LABEL_GENERATED} is not a state transition; it carries the generated
- * {@code parcelId} and is handled separately (TODO) rather than via the state machine.</p>
+ * <p>{@code parcelId} carries the generated barcode on {@code LABEL_GENERATED} (M4 stores it on the
+ * shipment, no state transition); it is null on the state-transition scans. {@code occurredAt} is
+ * the physical scan time. Tolerant reader — extra fields ignored (D-005).</p>
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (BanerjeeAgniva) [inline common/src/main/java/com/oneday/common/kafka/events/ScanEvent.java:17]: **The contract — not new event types.** This record is the message M8 publishes and M4 consumes on `oneday.scan.events`. PR2 adds two *fields* (`parcelId`, `occurredAt`), not new types (D-005).

Priya's flow: when her DA taps *Generate Label*, M8 emits `ScanEvent(S, LABEL_GENERATED, "1DD-DEL-260711-000042", now)`; M4's consumer reads `parcelId` and stamps it on the shipment. The 2-arg convenience ctor keeps PR3/PR4's state-transition scans terse; `@JsonIgnoreProperties` means older consumers just ignore the new fields.

File: barcode/src/main/java/com/oneday/barcode/api/ScanController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@RequestMapping("/api/v1/scan")
+class ScanController {
+
+    private final LabelService labelService;
+
+    ScanController(LabelService labelService) {
+        this.labelService = labelService;
+    }
+
+    @PostMapping("/label")
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/api/ScanController.java:24]: **The door the DA app knocks on.** Priya's DA reaches her doorstep, taps *Generate Label* → `POST /api/v1/scan/label`. This is M8's only write endpoint in PR2; the generic `/scan` for hub/airport/counter scans lands in PR3. Thin by design: validate → delegate to `LabelService` → return the barcode.

File: barcode/src/main/java/com/oneday/barcode/api/LabelRequest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import jakarta.validation.constraints.Size;
+
+import java.util.UUID;
+
+/**
+ * "Generate Label" from the DA app at pickup. {@code destCity} is the destination hub IATA (e.g.
+ * "DEL"); the app already shows it on the pickup task. {@code actorId} is the DA (ponytail: from the
+ * request until PR3 wires the auth principal). {@code clientScanId} makes the call idempotent.
+ */
+public record LabelRequest(
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/api/LabelRequest.java:14]: **What the DA app sends** at Priya's door: `{shipmentId, destCity:"DEL", actorId, clientScanId}`. `destCity` is the destination-hub IATA — the app already shows *Delhi* on the pickup task. Per **D-002** this is only the barcode's human-readable segment, never parsed for routing, so trusting the client for it is safe. `clientScanId` makes a flaky-signal retry idempotent. (`actorId` rides the request until PR3 wires the auth principal.)

File: barcode/src/main/java/com/oneday/barcode/api/LabelResponse.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.oneday.barcode.api;
+
+import java.util.UUID;
+
+/** The minted (or existing) barcode for the shipment — the DA app renders/prints {@code parcelId}. */
+public record LabelResponse(UUID shipmentId, String parcelId) {
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/api/LabelResponse.java:6]: **The sticker.** Returns `1DD-DEL-260711-000042` — the DA app renders it as a Code-128 and Priya's box wears it for the whole journey. Every later hub/van/airport scan keys off this exact string.

File: barcode/src/main/java/com/oneday/barcode/service/LabelService.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+public interface LabelService {
+
+    /**
+     * @param shipmentId  the shipment being labelled
+     * @param destCity    destination hub IATA (e.g. "DEL") — the barcode's human-readable segment
+     * @param actorId     the DA who scanned (recorded as the ledger actor)
+     * @param clientScanId device idempotency key (may be null)
+     * @return the barcode string, e.g. {@code 1DD-DEL-260711-000042}
+     */
+    String generateLabel(UUID shipmentId, String destCity, UUID actorId, UUID clientScanId);
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/LabelService.java:19]: **The "Generate Label" business action**, as one method. The whole first-mile-label story lives here: mint the barcode, write the `LABEL_GENERATED` scan, and (after commit) emit the `ScanEvent` M4 uses to fill `shipments.parcel_id`. Idempotent per shipment — a double-tap returns the same barcode.

File: barcode/src/main/java/com/oneday/barcode/service/LabelServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    LabelServiceImpl(ScanLedgerRepository repository, ParcelIdGenerator generator, ScanLedgerService ledger) {
+        this.repository = repository;
+        this.generator = generator;
+        this.ledger = ledger;
+    }
+
+    @Override
+    @Transactional
+    public String generateLabel(UUID shipmentId, String destCity, UUID actorId, UUID clientScanId) {
+        // Idempotent: a shipment is labelled once. A retry returns the existing barcode and — crucially
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/LabelServiceImpl.java:27]: **Idempotency is the load-bearing bit here.** If Priya's DA double-taps or retries on bad signal, we must NOT mint a second barcode *or* burn a counter value. So: look for an existing `LABEL_GENERATED` scan for this shipment first → return that barcode untouched. Only on the true first label do we call the generator + write the scan. `destCity` is trimmed/upper-cased (` del ` → `DEL`).

File: barcode/src/main/java/com/oneday/barcode/service/ParcelIdGenerator.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    // ponytail: hardcoded IST; all 5 launch cities are in one zone. Make configurable if we go global.
+    private static final ZoneId IST = ZoneId.of("Asia/Kolkata");
+
+    private final ParcelIdCounterRepository counters;
+
+    ParcelIdGenerator(ParcelIdCounterRepository counters) {
+        this.counters = counters;
+    }
+
+    /** Next barcode for a dest hub, for today (IST). Bumps the counter; call once per label. */
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/ParcelIdGenerator.java:30]: **Where `1DD-DEL-260711-000042` is born.** The `000042` is "the 42nd Delhi-bound parcel today", pulled from the per-hub/day counter under a `SELECT FOR UPDATE`, so two DAs printing at the same instant can't collide. `insertIfAbsent` materialises the `(DEL, today)` row (ON CONFLICT DO NOTHING), then read-and-increment under the lock, flushed on commit. Same pattern as `orders.ShipmentRefService`.

File: barcode/src/main/java/com/oneday/barcode/service/ScanLedgerService.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * The append-only write engine (M8 §4.3) — built once, reused by every scan entry door: the label
+ * path (PR2), the REST lifecycle scans (PR3), and the van custody port (PR4). Idempotent on
+ * {@code clientScanId}; emits a {@code ScanEvent} after commit for broadcastable scan types.
+ */
+public interface ScanLedgerService {
+
+    /** Append one immutable scan row (idempotent replay on {@code clientScanId}). */
+    ScanLedgerEntry record(ScanCommand cmd);
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/ScanLedgerService.java:13]: **The write engine — built once, used by all three entry doors.** PR2's label path is the first caller; PR3 (REST hub/airport scans) and PR4 (van custody port) reuse this exact `record(...)`. Centralising it means idempotency + the after-commit event are written once and inherited for free.

File: barcode/src/main/java/com/oneday/barcode/service/ScanLedgerServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    private final ApplicationEventPublisher events;
+
+    ScanLedgerServiceImpl(ScanLedgerRepository repository, ApplicationEventPublisher events) {
+        this.repository = repository;
+        this.events = events;
+    }
+
+    @Override
+    @Transactional
+    public ScanLedgerEntry record(ScanCommand cmd) {
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/ScanLedgerServiceImpl.java:23]: **"The scan physically happened, so record it."** Priya's label event becomes one immutable `scan_ledger` row. Replay guard first: if the same `clientScanId` already landed, return the existing row (no duplicate). After the insert it raises an in-process `ScanRecorded` — the outbound Rabbit event fires only AFTER this tx commits, so the event never races ahead of the durable truth. `@Transactional` holds the counter lock (from the generator) to commit.

File: barcode/src/main/java/com/oneday/barcode/service/ScanCommand.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.time.Instant;
+import java.util.UUID;
+
+/**
+ * One physical scan to append to the ledger (M8 §4) — the single input to {@link ScanLedgerService}.
+ * {@code parcelId}/{@code bagId} are set per scan family (barcode string on parcel scans, bag id on
+ * bag scans, both null on van scans). {@code clientScanId} is the device idempotency key.
+ */
+public record ScanCommand(
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/service/ScanCommand.java:11]: **The single input to the write engine.** One shape for every scan family: parcel scans (Priya's label) fill `parcelId`; bag scans (PR3) fill `bagId`; van scans (PR4) fill neither and travel by `shipmentId`. Keeps `record(...)` from needing a per-caller overload.

File: barcode/src/main/java/com/oneday/barcode/events/ScanRecorded.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.time.Instant;
+import java.util.UUID;
+
+/**
+ * In-process Spring event raised inside the ledger-write transaction. {@link ScanEventProducer}
+ * consumes it AFTER_COMMIT and decides whether to broadcast an outbound {@code ScanEvent} — the
+ * ledger is the durable truth, the event is a downstream signal.
+ */
+public record ScanRecorded(UUID shipmentId, String parcelId, String scanType, Instant scannedAt) {
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/events/ScanRecorded.java:11]: **In-process signal, not the Rabbit event.** Raised inside the write transaction; `ScanEventProducer` consumes it AFTER_COMMIT and decides whether to broadcast. This is the seam that guarantees *ledger row commits, THEN event publishes* — the same in-process→AFTER_COMMIT pattern M4's `ShipmentEventProducer` uses.

File: barcode/src/main/java/com/oneday/barcode/events/ScanEventProducer.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+    private static final Logger log = LoggerFactory.getLogger(ScanEventProducer.class);
+
+    private final EventPublisher eventPublisher;
+
+    ScanEventProducer(EventPublisher eventPublisher) {
+        this.eventPublisher = eventPublisher;
+    }
+
+    @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/events/ScanEventProducer.java:29]: **The AFTER_COMMIT publish.** Once Priya's ledger row is durable, this emits `ScanEvent(LABEL_GENERATED, parcelId)` to `oneday.scan.events` → M4 stamps `shipments.parcel_id`. Key rule: it broadcasts *only* types that are a `ScanEventType`. Van custody scans (`VAN_LOAD`…) aren't in that enum → `valueOf` throws → skipped (**D-004**: van scans are ledger-only; routing owns their state). This `try/valueOf` is exactly the split between the two scan families.

File: barcode/src/main/java/com/oneday/barcode/events/BarcodeMessagingConfig.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Declares M8's producing exchange so publishes aren't dropped before any consumer binds. Idempotent
+ * with orders' consumer-side declaration of the same durable topic exchange (RabbitAdmin dedups).
+ */
+@Configuration
+class BarcodeMessagingConfig {
+
+    @Bean
+    Declarables scanEventsExchange() {
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/events/BarcodeMessagingConfig.java:17]: **Declares M8's producing exchange** (`oneday.scan.events`) so a publish isn't dropped if M8 boots before any consumer binds. Idempotent with orders' consumer-side declaration of the same durable topic exchange — RabbitAdmin dedups identical declarations.

File: barcode/src/main/java/com/oneday/barcode/repository/ScanLedgerRepository.java [hunk 14]
```diff
@@ -4,6 +4,7 @@
 import org.springframework.data.repository.Repository;
 
 import java.util.List;
+import java.util.Optional;
```
Comment (BanerjeeAgniva) [inline barcode/src/main/java/com/oneday/barcode/repository/ScanLedgerRepository.java:7]: **Two new finders, both for idempotency** (added just below this import). `findByClientScanId` = the device-retry replay guard used by the write engine. `findFirstByShipmentIdAndScanType` = the label guard — "has Priya's shipment already been labelled?" → return that barcode instead of re-minting. Still no `delete`/`save`-as-update surfaced: the ledger stays append-only.

File: barcode/pom.xml [hunk 15]
```diff
@@ -24,6 +24,14 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (BanerjeeAgniva) [inline barcode/pom.xml:28]: **+amqp, +validation.** `amqp` because PR2 now *publishes* (the `ScanEventProducer` + exchange declaration) — PR1 was DB-only. `validation` for `@Valid` on the label request. Deliberately no barcode-imaging library: the DA app renders the Code-128 from the string M8 returns (**D-003**).

File: orders/src/main/java/com/oneday/orders/events/ScanEventsConsumer.java [hunk 16]
```diff
@@ -1,13 +1,16 @@
 package com.oneday.orders.events;
 
 import com.oneday.common.domain.enums.ShipmentState;
+import com.oneday.common.kafka.enums.ScanEventType;
```
Comment (BanerjeeAgniva) [inline orders/src/main/java/com/oneday/orders/events/ScanEventsConsumer.java:4]: **The M4 side of the seam** (new `ScanEventType` import; the handling is added below). Closes the pre-existing TODO: on `LABEL_GENERATED`, stamp `shipments.parcel_id` — **no** state transition (Priya's parcel doesn't change lifecycle state just because it got a sticker). Every other scan type still drives the state machine. A null `parcelId` or unknown shipment is logged, not thrown, so a bad event can't wedge the `orders.scan` queue.

File: orders/src/test/java/com/oneday/orders/events/ScanEventsConsumerTest.java [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * The M8 seam: LABEL_GENERATED stamps the barcode on the shipment (no state transition); every other
+ * scan type drives the state machine and never touches parcel_id.
+ */
+class ScanEventsConsumerTest {
+
+    private final ShipmentStateMachine stateMachine = mock(ShipmentStateMachine.class);
+    private final ShipmentRepository shipmentRepository = mock(ShipmentRepository.class);
+    private final ScanEventsConsumer consumer = new ScanEventsConsumer(stateMachine, shipmentRepository);
+
```
Comment (BanerjeeAgniva) [inline orders/src/test/java/com/oneday/orders/events/ScanEventsConsumerTest.java:26]: **Guards the M4 seam** (plain Mockito, mirrors `DaEventsConsumerTest`): `LABEL_GENERATED` stamps `parcel_id` + saves + no transition; `HUB_ORIGIN_IN` transitions + never stamps; a null-parcelId label is ignored. This is what proves Priya's `parcel_id` would get stamped — independent of the broker.

File: barcode/src/test/java/com/oneday/barcode/service/LabelServiceImplTest.java [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+class LabelServiceImplTest {
+
+    @Mock ScanLedgerRepository repository;
+    @Mock ParcelIdGenerator generator;
+    @Mock ScanLedgerService ledger;
+    @InjectMocks LabelServiceImpl service;
+
+    final UUID shipment = UUID.randomUUID();
+
+    @Test
```
Comment (BanerjeeAgniva) [inline barcode/src/test/java/com/oneday/barcode/service/LabelServiceImplTest.java:30]: **Guards the money path — idempotency.** Already-labelled shipment → same barcode, and `generator`/`ledger` are never touched (no counter bump). First label → mints, records `LABEL_GENERATED`, and normalises ` del ` → `DEL`. If someone later "optimises" away the existence check, this test goes red.

File: barcode/src/test/java/com/oneday/barcode/service/ParcelIdGeneratorTest.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+@ExtendWith(MockitoExtension.class)
+class ParcelIdGeneratorTest {
+
+    @Mock ParcelIdCounterRepository counters;
+    @InjectMocks ParcelIdGenerator generator;
+
+    @Test
+    void mintsFormattedBarcode_andBumpsCounter() {
+        ParcelIdCounter counter = new ParcelIdCounter();
```
Comment (BanerjeeAgniva) [inline barcode/src/test/java/com/oneday/barcode/service/ParcelIdGeneratorTest.java:30]: **Guards the format + counter maths:** `1DD-DEL-{today}-000042` (6-digit zero-pad) and the counter advances 42 → 43, with `insertIfAbsent` called first. The exact business promise of the barcode's `seq6`.

File: barcode/src/test/java/com/oneday/barcode/events/ScanEventProducerTest.java [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+import static org.mockito.Mockito.verifyNoInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class ScanEventProducerTest {
+
+    @Mock EventPublisher publisher;
+    @InjectMocks ScanEventProducer producer;
+
+    @Test
+    void broadcasts_whenScanTypeIsAScanEventType() {
```
Comment (BanerjeeAgniva) [inline barcode/src/test/java/com/oneday/barcode/events/ScanEventProducerTest.java:29]: **Guards the broadcast rule (D-004):** `LABEL_GENERATED` → published to `oneday.scan.events` carrying the barcode; `VAN_LOAD` → nothing published (ledger-only). This locks in that van custody scans never reach M4.

File: barcode/src/test/java/com/oneday/barcode/events/ScanEventJacksonTest.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+ * regression from the record's extra constructor.
+ */
+class ScanEventJacksonTest {
+
+    private final JsonMapper mapper = JsonMapper.builder()
+            .propertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
+            .addModule(new JavaTimeModule())
+            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+            .build();
+
```
Comment (BanerjeeAgniva) [inline barcode/src/test/java/com/oneday/barcode/events/ScanEventJacksonTest.java:28]: **This test exists because of a real bug hunt during boot-verify.** The async M4 stamp was landing in the DLQ; I suspected the record's extra 2-arg ctor broke Jackson's creator detection. It didn't — this proves `ScanEvent` round-trips through the app's Rabbit mapper (SNAKE_CASE + JavaTime/jsr310), so the consumer can deserialize exactly what M8 publishes. (Root cause turned out to be a pre-PR2 deployed consumer on the shared broker — see the PR description.)
